python3/httplib2/__init__.py

```diff
@@ -185,6 +185,13 @@ def _build_ssl_context(
         ssl.CERT_NONE if disable_ssl_certificate_validation else ssl.CERT_REQUIRED
     )
 
+    # SSLContext.maximum_version and SSLContext.minimum_version are python 3.7+.
+    # source: https://docs.python.org/3/library/ssl.html#ssl.SSLContext.maximum_version
+    if maximum_version is not None and hasattr(context, "maximum_version"):
```

Comment:
Can you also add code coverage?

Reply:
Travis Python 3.7 uses OpenSSL 1.0 which won't support these attributes, the only way to do testing would be to pre-compile Python and OpenSSL on Travis and then run the tests against that. I can submit such a PR for .travis.yml (I do something like this for my own downstream project) but it will slow Travis testing down somewhat on that configuration.

Reply:
@temoto Use tox-travis to force py37 and OpenSSL 1.1.0+? https://stackoverflow.com/questions/37016688/how-to-run-tox-with-travis-ci/

Reply:
My suggestion to keep things organized is either:
- full revamp of current config to take full advantage of tox-travis
- just another job conforming to existing config

```
jobs:
  include:
#...
    - stage: test
      env: _=py3-openssl-110
      python: 3.7
      install:
        - CUSTOM_OPENSSL_INSTALL_SCRIPT
        - pip install $pip_install_common 'codecov>=2.0.15' -r requirements-test.txt
      script: script/test -sv
```

I'd only slightly prefer not tox-travis route because it's many changes and you still need to install custom openssl.

CUSTOM_OPENSSL_INSTALL_SCRIPT should be noop with hot Travis cache.

Reply:
pyopenssl actually tests something like this on Travis using MacOS which can use OpenSSL 1.1.1b (latest) via a brew update:

https://travis-ci.org/pyca/pyopenssl/jobs/537306843

I'm currently testing adding an additional job that would use MacOS, Python 3.7 and OpenSSL 1.1+ so we can actually test TLS min/max availability.

Reply:
Ok, MacOS proved far to problematic so went with your suggestion of a custom script to compile.
- I've added tests that are now passing
- I removed Python 3.4 support as it's end of life and broke the tests.
Let me know if you have thoughts on better testing or anything else.